lib/parser-stats.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const promisify = require('bluebird').promisify;
+const Table = require('cli-table');
+const glob = promisify(require('glob').glob);;
+const filesize = require('filesize');
+const percent = require('calc-percent');
+const fs = require('fs');
+
+
+var table = new Table({
+	head: ['files', 'original size', 'optimized size', 'saving'],
+});
+
+function stats(config) {
+	return Promise.all(
+			Object.keys(config.parsers).map(function(name){
+				return Promise.all([
+					getSumOfFileSize(config.src, config.parsers[name].pattern),
+					getSumOfFileSize(config.temp, config.parsers[name].pattern)
+				])
+			})
+		).then(function (sizes) {
+			let sumSrc = 0;
+			let sumTemp = 0;
+			sizes.map(function(parsers){
+				let row = [];
+				let src = parsers[0];
+				let temp = parsers[1];
+				let diff = src.sum - temp.sum;
+				row.push(src.pattern);
+				row.push(filesize(src.sum));
+				sumSrc += src.sum;
+				row.push(filesize(temp.sum));
+				sumTemp += temp.sum;
+				row.push(percent(diff, [src.sum], {suffix: '%'}));
+				table.push(row);
+			});
+
+			let totalDiff = percent((sumSrc-sumTemp), sumSrc, {suffix: '%'});
+
+			table.push(['Total:', filesize(sumSrc), filesize(sumTemp), totalDiff]);
+
+			console.log(table.toString());
```

Comment:
If you return / resolve with `table.toString()` and `console.log` in the file calling `stats()` this module has one less thing it does. The module calling `stats()` could then instead of logging it out also save it to a file or whatever it wants with it.